document/js-api/index.bs

```diff
@@ -1470,25 +1473,80 @@ WebAssembly Module Records have the following methods:
 <h3 id="module-execution">ExecuteModule ( [ |promiseCapability| ] ) Concrete Method</h3>
 1. Assert: |promiseCapability| was not provided.
 1. Let |record| be this WebAssembly Module Record.
-1. Let |module| be |record|.\[[ModuleSource]].
-1. Let |imports| be a new, empty [=map=].
-1. For each (|importedModuleName|, |name|, <var ignore>type</var>) in [=module_imports=](|module|.\[[Module]]),
-    1. If |imports|[|importedModuleName|] does not exist, set |imports|[|importedModuleName|] to a new, empty [=map=].
+1. Let |module| be |record|.\[[ModuleSource]].\[[Module]].
+1. Let |imports| be « ».
+1. [=list/iterate|For each=] (|importedModuleName|, |name|, |importtype|) in [=module_imports=](|module|),
     1. Let |importedModule| be [$GetImportedModule$](|record|, |importedModuleName|).
-    1. Let |value| be ? |importedModule|.\[[Environment]].GetBindingValue(|name|, true).
-    1. Set |imports|[|importedModuleName|][|name|] to |value|.
-1. Let |importObject| be ! [$OrdinaryObjectCreate$](null).
-1. For each |key| → |value| of |imports|,
-    1. Let |moduleImportsObject| be ! [$OrdinaryObjectCreate$](null).
-    1. For each |importedName| → |importedValue| of |value|,
-        1. Perform ! [$CreateDataPropertyOrThrow$](|moduleImportsObject|, |importedName|, |importedValue|).
-    1. Perform ! [$CreateDataPropertyOrThrow$](|importObject|, |key|, |moduleImportsObject|).
-1. [=Read the imports=] of |module| with imports |importObject|, and let |imports| be the result.
+    1. Let |resolution| be |importedModule|.ResolveExport(|name|).
+    1. If |resolution|.\[[Module]] is a WebAssembly Module Record,
+        1. Let |resolutionInstance| be |resolution|.\[[Module]].\[[Instance]].
+        1. If |resolutionInstance| is ~empty~ then,
+            1. Throw a {LinkError} exception.
+        1. Let |resolutionModule| be |resolution|.\[[Module]].\[[ModuleSource]].\[[Module]].
```

Comment:
(this is confusing, I thought [[Module]] was the Module Record corresponding to [[ModuleSource]] rather than a WebAssembly module, but I don't think we can improve it anyway)

Reply:
Yes it's unfortunate we have `[[Module]]` on both a resolve binding record and a WebAssembly.Module module object. Perhaps resolve binding records could refactor at some point to use `[[ResolvedModule]]` or something like that to ensure clarity in this kind of scenario.

I've split out `resolutionModule` and `resolutionName` and also added some assertions to make the relationships clearer.